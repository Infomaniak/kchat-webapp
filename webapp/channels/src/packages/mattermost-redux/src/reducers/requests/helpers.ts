// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {AnyAction} from 'redux';

import type {RequestStatusType} from '@mattermost/types/requests';

import {RequestStatus} from 'mattermost-redux/constants';

export function initialRequestState(): RequestStatusType {
    return {
        status: RequestStatus.NOT_STARTED,
        error: null,
    };
}

export function handleRequest(
    REQUEST: string,
    SUCCESS: string,
    FAILURE: string,
    state: RequestStatusType,
    action: AnyAction,
): RequestStatusType {
    switch (action.type) {
    case REQUEST:
        return {
            ...state,
            status: RequestStatus.STARTED,
        };
    case SUCCESS:
        return {
            ...state,
            status: RequestStatus.SUCCESS,
            error: null,
        };
    case FAILURE: {
        return {
            ...state,
            status: RequestStatus.FAILURE,
            error: action.error,
        };
    }
    default:
        return state;
    }
}
