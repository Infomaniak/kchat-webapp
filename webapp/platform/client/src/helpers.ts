/* eslint-disable no-underscore-dangle */
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
export function buildQueryString(parameters: Record<string, any>): string {
    const keys = Object.keys(parameters);
    if (keys.length === 0) {
        return '';
    }

    const queryParams = Object.entries(parameters).
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        filter(([_, value]) => value !== undefined).
        map(([key, value]) => `${key}=${encodeURIComponent(value)}`).
        join('&');

    return queryParams.length > 0 ? `?${queryParams}` : '';
}

export function setUserAgent(window: Window, userAgent: string) {
    try {
        // Works on Firefox, Chrome, Opera and IE9+
        // @ts-expect-error depends on the browser
        if ((navigator as Window['navigator']).__defineGetter__) {
            // @ts-expect-error depends on the browser
            (navigator as Window['navigator']).__defineGetter__('userAgent', () => {
                return userAgent;
            });
        } else if (Object.defineProperty) {
            Object.defineProperty(navigator, 'userAgent', {
                get() {
                    return userAgent;
                },
            });
        }

        // Works on Safari
        if (window.navigator.userAgent !== userAgent) {
            const userAgentProp = {
                get() {
                    return userAgent;
                },
            };
            try {
                Object.defineProperty(window.navigator, 'userAgent', userAgentProp);
            } catch (e) {
                (window.navigator as Window['navigator']) = Object.create(navigator, {
                    userAgent: userAgentProp,
                });
            }
        }
    } catch (e) {
        // eslint-disable-next-line no-console
        console.error('[app/debug] unable to set userAgent.', e);
    }
}
