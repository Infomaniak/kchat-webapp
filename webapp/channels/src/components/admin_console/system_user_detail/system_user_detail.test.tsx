// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import type {RouteComponentProps} from 'react-router-dom';

import type {UserProfile} from '@mattermost/types/users';

import SystemUserDetail, {getUserAuthenticationTextField} from 'components/admin_console/system_user_detail/system_user_detail';
import type {
    Props,
    Params,
} from 'components/admin_console/system_user_detail/system_user_detail';

import {shallowWithIntl, type MockIntl} from 'tests/helpers/intl-test-helper';

describe('SystemUserDetail', () => {
    const defaultProps: Props = {
        mfaEnabled: false,
        patchUser: jest.fn(),
        updateUserMfa: jest.fn(),
        getUser: jest.fn(),
        updateUserActive: jest.fn(),
        setNavigationBlocked: jest.fn(),
        addUserToTeam: jest.fn(),
        openModal: jest.fn(),
        intl: {
            formatMessage: jest.fn(),
        } as MockIntl,
        ...({
            match: {
                params: {
                    user_id: 'user_id',
                },
            },
        } as RouteComponentProps<Params>),
    };

    test('should match default snapshot', () => {
        const props = defaultProps;
        const wrapper = shallowWithIntl(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot if MFA is enabled', () => {
        const props = {
            ...defaultProps,
            mfaEnabled: true,
        };
        const wrapper = shallowWithIntl(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('getUserAuthenticationTextField', () => {
    const intl = {formatMessage: ({defaultMessage}) => defaultMessage} as MockIntl;

    it('should return empty string if user is not provided', () => {
        const result = getUserAuthenticationTextField(intl, false, undefined);
        expect(result).toEqual('');
    });

    it('should return email if user has no auth service and MFA is not enabled', () => {
        const result = getUserAuthenticationTextField(intl, false, {auth_service: '', mfa_active: false} as UserProfile);
        expect(result).toEqual('Email');
    });

    it('should return auth service in uppercase if it is LDAP or SAML', () => {
        const result = getUserAuthenticationTextField(intl, false, {auth_service: 'ldap', mfa_active: false} as UserProfile);
        expect(result).toEqual('LDAP');
    });

    it('should return auth service in title case if it is not LDAP or SAML', () => {
        const result = getUserAuthenticationTextField(intl, true, {auth_service: 'oauth', mfa_active: false} as UserProfile);
        expect(result).toEqual('Oauth');
    });

    it('should include MFA if user has MFA enabled', () => {
        const result = getUserAuthenticationTextField(intl, true, {auth_service: 'oauth', mfa_active: true} as UserProfile);
        expect(result).toEqual('Oauth, MFA');
    });
});
