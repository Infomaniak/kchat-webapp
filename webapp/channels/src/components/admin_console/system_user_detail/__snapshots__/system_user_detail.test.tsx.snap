// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemUserDetail should match default snapshot 1`] = `
<div
  className="SystemUserDetail wrapper--fixed"
>
  <AdminHeader
    withBackButton={true}
  >
    <div>
      <Connect(Component)
        className="fa fa-angle-left back"
        to="/admin_console/user_management/users"
      />
      <MemoizedFormattedMessage
        defaultMessage="User Configuration"
        id="admin.systemUserDetail.title"
      />
    </div>
  </AdminHeader>
  <div
    className="admin-console__wrapper"
  >
    <div
      className="admin-console__content"
    >
      <AdminUserCard
        body={
          <React.Fragment>
            <span>
              
            </span>
            <label>
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Email"
                id="admin.userManagement.userDetail.email"
              />
              <Memo(EmailIcon) />
              <input
                className="form-control"
                disabled={false}
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
            <label>
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Username"
                id="admin.userManagement.userDetail.username"
              />
              <AtIcon />
              <span />
            </label>
            <label>
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Authentication Method"
                id="admin.userManagement.userDetail.authenticationMethod"
              />
              <ShieldOutlineIcon />
              <span>
                
              </span>
            </label>
          </React.Fragment>
        }
        footer={
          <React.Fragment>
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Reset Password"
                id="admin.user_item.resetPwd"
              />
            </button>
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Activate"
                id="admin.user_item.makeActive"
              />
            </button>
          </React.Fragment>
        }
        isLoading={true}
      />
      <AdminPanel
        button={
          <div
            className="add-team-button"
          >
            <button
              className="btn btn-primary"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Add Team"
                id="admin.userManagement.userDetail.addTeam"
              />
            </button>
          </div>
        }
        subtitle={
          Object {
            "defaultMessage": "Teams to which this user belongs",
            "id": "admin.userManagement.userDetail.teamsSubtitle",
          }
        }
        title={
          Object {
            "defaultMessage": "Team Membership",
            "id": "admin.userManagement.userDetail.teamsTitle",
          }
        }
      >
        <div
          className="teamlistLoading"
        >
          <Memo(LoadingSpinner) />
        </div>
      </AdminPanel>
    </div>
  </div>
  <div
    className="admin-console-save"
  >
    <SaveButton
      btnClass="btn-primary"
      defaultMessage={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Save"
          id="save_button.save"
        />
      }
      disabled={true}
      extraClasses=""
      onClick={[Function]}
      saving={false}
      savingMessage={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Saving"
          id="save_button.saving"
        />
      }
    />
    <div
      className="error-message"
    >
      <FormError
        error={null}
        errors={Array []}
      />
    </div>
  </div>
  <ConfirmModal
    confirmButtonClass="btn btn-danger"
    confirmButtonText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Deactivate"
        id="deactivate_member_modal.deactivate"
      />
    }
    message={
      <div>
        <Memo(MemoizedFormattedMessage)
          defaultMessage="This action deactivates {username}. They will be logged out and not have access to any teams or channels on this system. Are you sure you want to deactivate {username}?"
          id="deactivate_member_modal.desc"
          values={
            Object {
              "username": "",
            }
          }
        />
        <strong>
          <br />
          <br />
          <Memo(MemoizedFormattedMessage)
            defaultMessage="You must also deactivate this user in the SSO provider or they will be reactivated on next login or sync."
            id="deactivate_member_modal.sso_warning"
          />
        </strong>
      </div>
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Deactivate {username}"
        id="deactivate_member_modal.title"
        values={
          Object {
            "username": "",
          }
        }
      />
    }
  />
</div>
`;

exports[`SystemUserDetail should match snapshot if MFA is enabled 1`] = `
<div
  className="SystemUserDetail wrapper--fixed"
>
  <AdminHeader
    withBackButton={true}
  >
    <div>
      <Connect(Component)
        className="fa fa-angle-left back"
        to="/admin_console/user_management/users"
      />
      <MemoizedFormattedMessage
        defaultMessage="User Configuration"
        id="admin.systemUserDetail.title"
      />
    </div>
  </AdminHeader>
  <div
    className="admin-console__wrapper"
  >
    <div
      className="admin-console__content"
    >
      <AdminUserCard
        body={
          <React.Fragment>
            <span>
              
            </span>
            <label>
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Email"
                id="admin.userManagement.userDetail.email"
              />
              <Memo(EmailIcon) />
              <input
                className="form-control"
                disabled={false}
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
            <label>
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Username"
                id="admin.userManagement.userDetail.username"
              />
              <AtIcon />
              <span />
            </label>
            <label>
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Authentication Method"
                id="admin.userManagement.userDetail.authenticationMethod"
              />
              <ShieldOutlineIcon />
              <span>
                
              </span>
            </label>
          </React.Fragment>
        }
        footer={
          <React.Fragment>
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Reset Password"
                id="admin.user_item.resetPwd"
              />
            </button>
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Activate"
                id="admin.user_item.makeActive"
              />
            </button>
          </React.Fragment>
        }
        isLoading={true}
      />
      <AdminPanel
        button={
          <div
            className="add-team-button"
          >
            <button
              className="btn btn-primary"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Add Team"
                id="admin.userManagement.userDetail.addTeam"
              />
            </button>
          </div>
        }
        subtitle={
          Object {
            "defaultMessage": "Teams to which this user belongs",
            "id": "admin.userManagement.userDetail.teamsSubtitle",
          }
        }
        title={
          Object {
            "defaultMessage": "Team Membership",
            "id": "admin.userManagement.userDetail.teamsTitle",
          }
        }
      >
        <div
          className="teamlistLoading"
        >
          <Memo(LoadingSpinner) />
        </div>
      </AdminPanel>
    </div>
  </div>
  <div
    className="admin-console-save"
  >
    <SaveButton
      btnClass="btn-primary"
      defaultMessage={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Save"
          id="save_button.save"
        />
      }
      disabled={true}
      extraClasses=""
      onClick={[Function]}
      saving={false}
      savingMessage={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Saving"
          id="save_button.saving"
        />
      }
    />
    <div
      className="error-message"
    >
      <FormError
        error={null}
        errors={Array []}
      />
    </div>
  </div>
  <ConfirmModal
    confirmButtonClass="btn btn-danger"
    confirmButtonText={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Deactivate"
        id="deactivate_member_modal.deactivate"
      />
    }
    message={
      <div>
        <Memo(MemoizedFormattedMessage)
          defaultMessage="This action deactivates {username}. They will be logged out and not have access to any teams or channels on this system. Are you sure you want to deactivate {username}?"
          id="deactivate_member_modal.desc"
          values={
            Object {
              "username": "",
            }
          }
        />
        <strong>
          <br />
          <br />
          <Memo(MemoizedFormattedMessage)
            defaultMessage="You must also deactivate this user in the SSO provider or they will be reactivated on next login or sync."
            id="deactivate_member_modal.sso_warning"
          />
        </strong>
      </div>
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Deactivate {username}"
        id="deactivate_member_modal.title"
        values={
          Object {
            "username": "",
          }
        }
      />
    }
  />
</div>
`;
